account_financial_report_webkit/report/common_partner_reports.py

```diff
@@ -89,17 +89,20 @@ def _get_first_special_period(self):
     def _get_period_range_from_start_period(self, start_period,
                                             include_opening=False,
                                             fiscalyear=False,
-                                            stop_at_previous_opening=False):
+                                            stop_at_previous_opening=False,
+                                            **kwargs):
         """We retrieve all periods before start period"""
         periods = super(CommonPartnersReportHeaderWebkit, self).\
             _get_period_range_from_start_period(
                 start_period,
                 include_opening=include_opening,
                 fiscalyear=fiscalyear,
                 stop_at_previous_opening=stop_at_previous_opening)
-        first_special = self._get_first_special_period()
-        if first_special and first_special.id not in periods:
-            periods.append(first_special.id)
+
+        if not kwargs.get('exclude_first_special_period', False):
```

Comment:
When you exclude the first special period, this leads to _compute_init_balance being called with an empty list of periods which raises the exception in the tests. Isn't it safe to include the first special period, or rather: will the reports actually be correct without it?

Reply:
Yes, in context of `Partner Balance` report (and other reports) we need to exclude first special period as those numbers are gathered via `account.static.balance` `get_balances` method, so the initial balances gets duplicated and thus we have invalid report.

If I'm right here then we have problem with other reports as well not just with partner balance.
So, I added those to log lines to be sure that the context values from yml test file are correctly propagated to the report, because something tells me that those values are missing for some reason.

Let's see ....

Reply:
As i was afraid of, context values are not passed from yml test file to the report.
Full Log is [here](https://build.emadev.nl/blue/organizations/jenkins/BillingHouse%2FEssent%2FEssent-BE-Odoo-Jenkinsfile/detail/PR-1670/6/pipeline)

Below are debug lines from the Partner Balance report:
> [flow[60860166-b33f-44e1-b21b-ded430375b0d][test:set_metadata][Administrator][odoo]]2020-01-23 09:39:50,868 UTC 2127 INFO odoo_1 openerp.addons.account_financial_report_webkit.report.partner_balance: include_opening flag set to: 'value not present in context'
[flow[60860166-b33f-44e1-b21b-ded430375b0d][test:set_metadata][Administrator][odoo]]2020-01-23 09:39:50,868 UTC 2127 INFO odoo_1 openerp.addons.account_financial_report_webkit.report.partner_balance: `exclude_first_special_period` flag set to: 'value not present in context'
[flow[60860166-b33f-44e1-b21b-ded430375b0d][test:set_metadata][Administrator][odoo]]2020-01-23 09:39:50,869 UTC 2127 INFO odoo_1 openerp.addons.account_financial_report_webkit.report.partner_balance: `exclude_first_special_period` flag set to: 'value not present in context'

@StefanRijnhart  Do you have any idea why context from yml test is not propagated to the report?